gst/tensor_merge/gsttensormerge.c

```diff
@@ -0,0 +1,780 @@
+    GstTensorMergePadData * old, GstTensorMergePadData * new)
+{
+  guint64 oldtime, newtime;
+  if (old == NULL || old->buffer == NULL)
+    return 1;
+  if (new == NULL || new->buffer == NULL)
+    return -1;
+  if (GST_CLOCK_TIME_IS_VALID (old->dts_timestamp) &&
+      GST_CLOCK_TIME_IS_VALID (new->dts_timestamp)) {
+    oldtime = old->dts_timestamp;
+    newtime = new->dts_timestamp;
+  } else {
+    oldtime = old->pts_timestamp;
+    newtime = new->pts_timestamp;
+  }
+
+  if (!GST_CLOCK_TIME_IS_VALID (oldtime))
+    return -1;
+  if (!GST_CLOCK_TIME_IS_VALID (newtime))
+    return 1;
+
+  if (newtime < oldtime)
+    return 1;
+  else if (newtime > oldtime)
+    return -1;
+
+  return 0;
+}
+
+/**
+ * @brief Generate TensorConfig with TensorsConfig
+ * @param tensor_merge tensor merger
+ * @param configs Tensors Config Data
+ * @param config Tensor Config Data
+ * @return true / false
+ */
+gboolean
+gst_merge_tensors_config (GstTensorMerge * tensor_merge,
+    GstTensorsConfig * configs, GstTensorConfig * config)
+{
+  gboolean ret = FALSE;
+  switch (tensor_merge->mode) {
+    case GTT_LINEAR:
+    {
+      switch (tensor_merge->data_linear.direction) {
+          /* TODO : Currently we merge channel only. Have to extend W,H direction */
+        case LINEAR_CHANNEL:
+        {
+          int i, width, height, channel, batch;
+          tensor_type type;
+          type = configs->info.info[0].type;
+          batch = configs->info.info[0].dimension[3];
+          width = configs->info.info[0].dimension[1];
+          height = configs->info.info[0].dimension[2];
+          channel = configs->info.info[0].dimension[0];
+
+          for (i = 1; i < configs->info.num_tensors; i++) {
+            if (type != configs->info.info[i].type)
+              GST_ELEMENT_ERROR (tensor_merge, CORE, NEGOTIATION, (NULL),
+                  (NULL));
+            if (batch != configs->info.info[i].dimension[3])
+              GST_ELEMENT_ERROR (tensor_merge, CORE, NEGOTIATION, (NULL),
+                  (NULL));
+            if (width != configs->info.info[i].dimension[1])
+              GST_ELEMENT_ERROR (tensor_merge, CORE, NEGOTIATION, (NULL),
+                  (NULL));
+            if (height != configs->info.info[i].dimension[2])
+              GST_ELEMENT_ERROR (tensor_merge, CORE, NEGOTIATION, (NULL),
+                  (NULL));
+            channel += configs->info.info[i].dimension[0];
+          }
+
+          config->info.type = type;
+          config->info.dimension[0] = width;
+          config->info.dimension[1] = height;
+          config->info.dimension[2] = channel;
+          config->info.dimension[3] = batch;
+          config->rate_d = configs->rate_d;
+          config->rate_n = configs->rate_n;
+        }
+          ret = TRUE;
+          break;
+        case LINEAR_WIDTH:
+        case LINEAR_HEIGHT:
+        default:
+          ret = FALSE;
+      }
+    }
+      break;
+    default:
+      if (!ret)
```

Comment:
Please consider this line. 
If ret=GST_FLOW_NOT_SUPPORTED supposed error, but it will run as true.